crates/nargo/src/cli/compile_cmd.rs

```diff
@@ -25,17 +25,14 @@ pub(crate) fn run(args: ArgMatches) -> Result<(), CliError> {
     let mut circuit_path = PathBuf::new();
     circuit_path.push(TARGET_DIR);
 
-    let result = generate_circuit_and_witness_to_disk(
+    generate_circuit_and_witness_to_disk(
         circuit_name,
         curr_dir,
         circuit_path,
         witness,
         allow_warnings,
-    );
-    match result {
-        Ok(_) => Ok(()),
-        Err(e) => Err(e),
-    }
+    )
+    .map(|_| ())
```

Comment:
Not blocking(just a comment): Mapping to the unit value seems a bit out of place here, but perhaps run should simply return the PathBuf that we get from `generate_circuit_and_witness_to_disk`, then the caller ignores it.

Reply:
I only meant to clean this code a bit and did not change the previous behavior. I only moved the explicit match into a map. I don't think we should return a PathBuf for the sake of returning something.

Reply:
Agreed. Perhaps it was supposed to return a PathBuf from the beginning, I'll need to check the usage